src/rgw/rgw_cache.cc

```diff
@@ -284,9 +286,31 @@ void ObjectCache::set_enabled(bool status)
 
 void ObjectCache::invalidate_all()
 {
-  RWLock::WLocker l(lock);
+  std::unique_lock l(lock);
+
+  do_invalidate_all();
+}
+
+uint64_t ObjectCache::bump_epoch() {
+  std::unique_lock l(lock);
 
+  ++epoch;
   do_invalidate_all();
+  return epoch;
+}
+
+bool ObjectCache::handle_epoch(uint64_t e) {
+  std::unique_lock l(lock);
```

Comment:
@adamemerson this will serialize all requests here, whereas cache hardly ever changes. Maybe get a read lock first, and only if needed then reacquire a write lock?

Reply:
Thank you! That is an excellent point. I hadn't noticed the cache is read/write locked but it is.